packages/core/src/storages/utils.ts

```diff
@@ -22,3 +24,30 @@ export async function purgeDefaultStorages(config = Configuration.getGlobalConfi
         await client.purge?.();
     }
 }
+
+interface UseStateOptions {
+    storeManagerOptions?: StorageManagerOptions;
+    /**
+     * The name of the key-value store you'd like the state to be stored in.
+     * If not provided, the default store will be used.
+     */
+    keyValueStoreName?: string | null;
+}
+
+/**
+ * Easily create and manage state values. All state values are automatically persisted.
+ *
+ * Values can be modified by simply using the assignment operator.
+ *
+ * @param name The name of the store to use.
+ * @param defaultValue If the store does not yet have a value in it, the value will be initialized with the `defaultValue` you provide.
+ * @param options An optional object parameter where a custom `keyValueStoreName` and `storeManagerOptions` can be passed in.
+ */
+export async function useState<T extends Dictionary = Dictionary>(
+    name: string,
+    defaultValue = {} as T,
+    options?: UseStateOptions,
+) {
+    const kvStore = await KeyValueStore.open(options?.keyValueStoreName, options?.storeManagerOptions || { config: Configuration.getGlobalConfig() });
+    return kvStore.getAutoSavedValue<T>(name?.toUpperCase() || 'CRAWLEE_STATE', defaultValue);
```

Comment:
i dont think we need to convert the name anyhow, if user wants to have it lower case, why not?

i am also not 100% sure if we should use `CRAWLEE_STATE` as the default key -- its the same the `crawler.useState` uses, so you would override the value. or was that on purpose, so you have one state you can access inside the request handler too?

Reply:
uppercase ftw imo, but I'll remove it